source/docs/advanced/features/smart-forms.md

```diff
@@ -217,3 +217,102 @@ const MyAwesomeForm = ({
 ```
 
 After implementing the above changes, when the user submits the form it will validate the phone number first. Then if the phone number is valid the form will be submitted. Otherwise the form will not be submitted and the `validationError` variable above will be set to a translated message of `"Invalid phone number"`. (P.S. if for some the form validation fails for reasons other than the phone number being invalid, the form submit will be called as to not block the user).
+
+## `useEmailServerValidation`
+
+As its name suggests, `useEmailServerValidation` can be used to validate an email with the help of a server GraphQL API.
+
+### Basic usage
+
+When calling this hook, you have to pass three parameters:
+
+1. the name of the field where the user is suppose to enter an email
+1. a loading handling function. It will receive a boolean indicating when the server validation starts or ends
+1. a submit function. It will receive a hash with the field values. Typically, this function would call the GraphQL API to store the entity the form is about
+
+So on an existing form, you would do the following changes:
+
+```diff
+import Form from "theme/components/atoms/Form/Form";
+import { Email } from "theme/components/atoms/Form/Input";
++import { useEmailServerValidation } from "theme/modules/SmartForms/Field/Email";
+
+ const MyAwesomeForm = ({ email }) => {
+   const submit = (data) => {
+     // your submit implementation
+   };
++  const setLoading = (isLoading) {
++    // do something when the server starts or ends
++  };
++  const onSubmit = useEmailServerValidation("email", setLoading, submit);
+
+   return (
+-    <Form onValidSubmit={submit}>
++    <Form onValidSubmit={onSubmit}>
+        <Email name="email" id="email" value={email} required />
+        <!-- probably some other fields -->
+     </Form>
+```
+
+With this code, when the user submits the form, the hook will call the GraphQL API to validate the email. The GraphQL email validation API is designed [to return different validation statuses](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/546/diffs#e987e1da0e06e52d1b120c9d0081953f11fc8c7a_13_18) (not only valid/invalid) and by default, the `onSubmit` handler returned by `useEmailServerValidation` has the following behaviour:
```

Comment:
Nit: the MR will remain as is while the code will stay up-to-date. While it can also be refactored, in this context it seems more reliable than the MR
```suggestion
With this code, when the user submits the form, the hook will call the GraphQL API to validate the email. The GraphQL email validation API is designed [to return different validation statuses](https://gitlab.com/front-commerce/front-commerce/-/blob/main/src/server/modules/front-commerce/smart-forms/loaders/EmailValidation.js#L4) (not only valid/invalid) and by default, the `onSubmit` handler returned by `useEmailServerValidation` has the following behaviour:
```

Reply:
that makes sense but then maybe the link should point to the GraphQL schema which is probably the less likely to change